starter-code/pizza.js

```diff
@@ -17,69 +17,145 @@ var state = {
   greenPeppers: true,
   whiteSauce: false,
   glutenFreeCrust: false
-}
+};
 
 // This function takes care of rendering the pizza based on the state
 // This function is triggered once at the begining and everytime the state is changed
 function renderEverything() {
-  renderPepperonni()
-  renderMushrooms()
-  renderGreenPeppers()
-  renderWhiteSauce()
-  renderGlutenFreeCrust()
-
-  renderButtons()
-  renderPrice()
+  renderPepperonni();
+  renderMushrooms();
+  renderGreenPeppers();
+  renderWhiteSauce();
+  renderGlutenFreeCrust();
+
+  renderButtons();
+  renderPrice();
 }
 
 function renderPepperonni() {
-  document.querySelectorAll('.pep').forEach(function($pep){
+  document.querySelectorAll(".pep").forEach(function($pep) {
     if (state.pepperonni) {
       $pep.style.visibility = "visible";
-    }
-    else {
+    } else {
       $pep.style.visibility = "hidden";
     }
-  })
+  });
 }
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll(".mushroom").forEach(function($mush) {
+    if (state.mushrooms) {
+      $mush.style.visibility = "visible";
+    } else {
+      $mush.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll(".green-pepper").forEach(function($pepper) {
+    if (state.greenPeppers) {
+      $pepper.style.visibility = "visible";
+    } else {
+      $pepper.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll(".sauce").forEach(function($sauce) {
+    if (state.sauce) {
+      $sauce.style.visibility = "visible";
+    } else {
+      $sauce.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  document.querySelectorAll(".crust").forEach(function($crust) {
+    if (state.crust) {
```

Comment:
Same thing here. Basically if you set the visibility here to hidden, nothing appears. so if you dont have anything selected, there's no pizza to be seen, and that is not supposed to be the case.